site/en/docs/extensions/reference/runtime/index.md

```diff
@@ -2,4 +2,79 @@
 api: runtime
 ---
 
-<!-- Intentionally blank -->
+## Examples
+
+### Use [`getURL`][getURL] to add an extension image to a page {: #example-getURL }
+
+Extensions can access their own resources using relative paths, but in order for the extension's
+assets to appear in a web page, you will need to specify the full URL of the asset (and expose it
+using the [web_accessible_resources][war] field).
+
+This example shows how a [content script][content] could add an image in the extension's package to
+the page that the content script has been [injected][content-inject] into.
+
+```js
+//// content.js
+
+{ // Block used to avoid setting global variables
+  let img = document.createElement('img');
+  img.src = chrome.runtime.getURL('logo.png');
+  document.body.append(img);
+}
+```
+
+### Use runtime messaging to communicate between background and content scripts {: #example-background-content-messaging }
+
+A basic example of content script injected into a page and background script passing messages to
+each other. This exchange is modeled on a [TCP handshake][handshake].
+
+```js
+//// content.js ////
+
+// 1. Send the background a message requesting receipt confirmation.
+console.log('[content] sending content-syn');
+chrome.runtime.sendMessage('content-syn');
+
+chrome.runtime.onMessage.addListener((message, sender) => {
+  if (message === 'bg-syn-ack') {
+    // 3. BG syn-ack received, acknowledged receipt.
+    console.log('[content] received bg-syn-ack');
+    console.log('[content] sending content-ack');
+    chrome.runtime.sendMessage('content-ack');
+  }
+});
+```
+
+```js
+//// background.js ////
+
+chrome.runtime.onMessage.addListener((message, sender) => {
+  if (message.type === 'content-syn') {
+    // 2. Content syn received, acknowledge receipt & request confirmation.
+    console.log('[bg] received content-syn');
+    console.log('[bg] sending bg-syn-ack');
+    chrome.tabs.sendMessage(sender.tab.id, 'bg-syn-ack');
+  }
+  if (message.type === 'content-ack') {
+    // 4. Content ack received. Both ends have verified that message
+    //    passing works as expected.
+    console.log('[bg] received content-ack');
+  }
+});
+```
+
+### Getting an extension's key from it's manifest {: #example-key }
+
+Retrieve a production extension's [public key][key-prop] for use in development. Execute this snippet in a DevTools console for the production extension paste the value in your unpacked extension's manifest.json.
```

Comment:
Consider:

Sometimes in development you may need to use the production extension's [public key][key-prop]. The following example shows a snippet that retrieves the public key; you can execute this in the DevTools console...

BUT -- is "execute this in the DevTools console" the kind of example we want?

Reply:
That's a fair question. I've found this to be very useful, but I need to do more key manipulation than the average extension developer. I'm inclined to keep it for the moment, but I'm open to dropping it if we haver other examples that would make better use of the page space.

Reply:
On second thought, I'm not wild about encouraging people to copy-paste stuff into their console. I'm going to replace this example with something else.